semantic-python/src/Language/Python/Core.hs

```diff
@@ -141,7 +135,7 @@ instance Compile Py.Attribute
 -- RHS represents the right-hand-side of an assignment that we get out of tree-sitter.
 -- Desugared is the "terminal" node in a sequence of assignments, i.e. given a = b = c,
 -- c will be the terminal node. It is never an assignment.
-type RHS = Py.Assignment :+: Py.AugmentedAssignment :+: Desugared
+type RHS = (Py.Assignment :+: Py.AugmentedAssignment) :+: Desugared
```

Comment:
Sum trees are balanced now.

Reply:
I’m not sure I understand why this is necessary?

Reply:
We no longer generate sums as `A :+: B :+: C :+: D` in a right-nested chain (`:+:` is infixr); instead, they’re balanced binary trees: `(A :+: B) :+: (C :+: D)`, which is distinct from `A :+: B :+: (C :+: D)`.